fe/fe-core/src/main/java/com/starrocks/catalog/OlapTable.java

```diff
@@ -233,7 +234,7 @@ public enum OlapTableState {
     protected TableProperty tableProperty;
 
     @SerializedName(value = "maxColUniqueId")
-    protected int maxColUniqueId = -1;
+    protected AtomicInteger maxColUniqueId = new AtomicInteger(-1);
```

Comment:
This change is good to me, and it should be done in this way.

But I have a little concern that this field is persisted in the metadata and you are changing the type of this field. Have you verified that `maxColUniqueId` will be the right number after recovering from the image that the current version generated?

Reply:
> This change is good to me, and it should be done in this way.
> 
> But I have a little concern that this field is persisted in the metadata and you are changing the type of this field. Have you verified that `maxColUniqueId` will be the right number after recovering from the image that the current version generated?

In the function `gsonPostProcess`, we will rebuild  full schema and we will update the `maxColUniqueId` as the max unique id of all columns(the max unique id of all columns should be equal to maxColUniqueId). So we can recover from the image correctly whether upgrade or downgrade.